src/application/services/useTools.ts

```diff
@@ -63,20 +65,21 @@ export function useTools(noteTools: Ref<EditorTool[]>): {
   watch(
     [userEditorTools, noteTools],
     async () => {
-      const allTools = mergeTools(userEditorTools.value || [], noteTools.value);
+      allTools.value = mergeTools(userEditorTools.value || [], noteTools.value);
 
       /**
        * If tools are not loaded yet or empty, skip downloading their scripts
        */
-      if (allTools.length === 0) {
+      if (allTools.value.length === 0) {
         return;
       }
-      tools.value = await downloadTools(allTools);
+      tools.value = await downloadTools(allTools.value);
     },
     { immediate: true }
   );
 
   return {
     tools,
+    allTools,
```

Comment:
There is a problem with naming: 
seems like `tools` and `allTools` contains the same set. But first contains downloaded something, and all tools contains just info. 

I'w suggest:

`tools` -> `toolsConnected`
`allTools` -> `tools`